modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/entity/Bookmark.kt

```diff
@@ -16,13 +19,32 @@ import java.util.Date
 )
 data class Bookmark(
     @PrimaryKey(autoGenerate = false) @ColumnInfo(name = "uuid") var uuid: String,
-    @ColumnInfo(name = "podcast_uuid") var podcastUuid: String,
-    @ColumnInfo(name = "episode_uuid") var episodeUuid: String,
-    @ColumnInfo(name = "time") var timeSecs: Int,
-    @ColumnInfo(name = "created_at") var createdAt: Date,
-    @ColumnInfo(name = "title") var title: String,
+    @ColumnInfo(name = "podcast_uuid") var podcastUuid: String = "",
+    @ColumnInfo(name = "episode_uuid") var episodeUuid: String = "",
+    @ColumnInfo(name = "time") var timeSecs: Int = 0,
+    @ColumnInfo(name = "created_at") var createdAt: Date = Date(),
+    @ColumnInfo(name = "title") var title: String = "",
     @ColumnInfo(name = "title_modified") var titleModified: Long? = null,
     @ColumnInfo(name = "deleted") var deleted: Boolean = false,
     @ColumnInfo(name = "deleted_modified") var deletedModified: Long? = null,
-    @ColumnInfo(name = "sync_status") var syncStatus: SyncStatus
-) : Serializable
+    @ColumnInfo(name = "sync_status") var syncStatus: SyncStatus = SyncStatus.NOT_SYNCED,
+    @Ignore val episodeTitle: String = "",
+) : Serializable {
+    constructor() : this(uuid = "")
+
+    val adapterId: Long
+        get() = UUID.nameUUIDFromBytes(uuid.toByteArray()).mostSignificantBits
+
+    fun createdAtDatePattern(): String {
+        val calendar = Calendar.getInstance()
+        val currentYear = calendar.get(Calendar.YEAR)
+        calendar.time = createdAt
+        val createdAtYear = calendar.get(Calendar.YEAR)
+
+        return if (createdAtYear == currentYear) {
+            "MMM d 'at' h:mm a"
+        } else {
+            "MMM d, YYYY 'at' h:mm a"
```

Comment:
I wonder if we should exclude the `at` text as it doesn't get localised. 

<img width="257" alt="Screenshot 2023-07-27 at 5 56 11 pm" src="https://github.com/Automattic/pocket-casts-android/assets/308331/c4aef0a8-f261-4f85-abce-db32d44b7664">

Reply:
Removed in [24e74c7](https://github.com/Automattic/pocket-casts-android/pull/1210/commits/24e74c78d0f8961006bfd18b3b8b5e2cf09bdc2b)